### PR TITLE
fix Issue 18597 - more unsafe unaligned pointer errors

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/dstruct.d [hunk 1]
```diff
@@ -479,6 +479,19 @@ extern (C++) class StructDeclaration : AggregateDeclaration
             Type origType = t;
             Type tb = t.toBasetype();
 
+            const hasPointers = tb.hasPointers();
```
Comment (stefan-koch-sociomantic) [inline src/dmd/dstruct.d]: seems to me like this block would be big enough to put it into a function.
This would help preventing diverging error messages.

**Comment by WalterBright** [comment]
The autotester on Darwin_64_32 is seg faulting when compiling dmd. Is it using the same version of dmd as the other test platforms?

**Comment by wilzbach** [comment]
Maybe related: https://github.com/dlang/dmd/pull/8022#comment-373617747

**Comment by WalterBright** [comment]
ping @wilzbach

**Review by JinShil — APPROVED** [review]

**Pull request merged**